### PR TITLE
Improve Locale Chooser

### PR DESCRIPTION
**Pull request by jrjohnson**
Converted to glimmer component and octane patterns
Fixed a11y where the menubar and menuitem were incorrectly applied to the toggle.

### CODE REVIEW

File: app/components/locale-chooser.js [hunk 1]
```diff
@@ -1,81 +1,86 @@
...
       break;
     case 'ArrowUp':
-      if (originalEvent.target.previousElementSibling) {
-        originalEvent.target.previousElementSibling.focus();
+      event.preventDefault();
+      if (target.previousElementSibling) {
+        target.previousElementSibling.focus();
       } else {
-        this.element.querySelector('.menu button:last-of-type').focus();
+        this.menuElement.querySelector('button:last-of-type').focus();
```
Comment (stopfstedt) [inline app/components/locale-chooser.js]: should be `this.element...`

File: app/components/locale-chooser.js [hunk 2]
```diff
@@ -1,81 +1,86 @@
...
+        target.nextElementSibling.focus();
       } else {
-        if (originalEvent.target.nextElementSibling) {
-          originalEvent.target.nextElementSibling.focus();
-        } else {
-          schedule('afterRender', () => {
-            this.element.querySelector('.menu button:nth-of-type(1)').focus();
-          });
-        }
+        this.menuElement.querySelector('button:nth-of-type(1)').focus();
```
Comment (stopfstedt) [inline app/components/locale-chooser.js]: should be `this.element...`

File: app/components/locale-chooser.js [hunk 3]
```diff
@@ -1,81 +1,86 @@
...
     case 'ArrowRight':
     case 'ArrowLeft':
-      this.set('isOpen', false);
+      this.isOpen = false;
+      break;
+    }
+  }
+  @action
+  clearFocus() {
+    const buttons = this.menuElement.querySelectorAll('button');
```
Comment (stopfstedt) [inline app/components/locale-chooser.js]: should be `this.element...`

File: app/components/locale-chooser.js (hunk 1 above)
Comment (jrjohnson) [reply app/components/locale-chooser.js]: `this.element` doesn't exist for glimmer components, instead this uses the `{{ref}} modifier in the template to assign an element to `menuElement` https://guides.emberjs.com/release/components/template-lifecycle-dom-and-modifiers/#toc_communicating-between-elements-in-a-component

Comment (stopfstedt) [reply app/components/locale-chooser.js]: i stand corrected.

**Pull request merged**